pkg/codegen/go/gen_program.go

```diff
@@ -145,21 +146,34 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 	}
 	files["Pulumi.yaml"] = projectBytes
 
+	packages, err := programPackageDefs(program)
+	if err != nil {
+		return err
+	}
+
 	// Build a go.mod based on the packages used by program
 	var gomod bytes.Buffer
 	gomod.WriteString("module " + project.Name.String() + "\n")
+	gomod.WriteString("go 1.17\n")
+	for name, sdk := range localProjects {
+		if name == "go" {
```

Comment:
I've opted to use the language instead of "pulumi" for the language SDKs, because otherwise either 1. "pulumi" would point to different places for different tests (doable, imo slightly confusing but not the worst), or 2. "pulumi" would just point to p/p/sdk and then the rest of the path would have to be appended (makes assumptions about where the SDKs are). I think that it makes more sense to just do it this way, particularly if developing only one language SDK.

Reply:
Ah right but see comments further down that I think the matrix testing should be building this map per-language, and so both of these issues would be solved